src/coordinator/ts/m3db/block/iterator.go

```diff
@@ -0,0 +1,214 @@
+		return consolidatedSeriesBlockIters
+	}
+
+	for i, seriesBlock := range blocks {
+		consolidatedNSBlockIters := make([]block.ValueIterator, len(blocks[0].ConsolidatedNSBlocks))
+		for j, nsBlock := range seriesBlock.ConsolidatedNSBlocks {
+			nsBlockIter := newConsolidatedNSBlockIter(nsBlock)
+			consolidatedNSBlockIters[j] = nsBlockIter
+		}
+		consolidatedSeriesBlockIters[i] = &consolidatedSeriesBlockIter{
+			consolidatedNSBlockIters: consolidatedNSBlockIters,
+		}
+	}
+
+	return consolidatedSeriesBlockIters
+}
+
+func newConsolidatedNSBlockIter(nsBlock ConsolidatedNSBlock) *consolidatedNSBlockIter {
+	return &consolidatedNSBlockIter{
+		consolidatedNSBlockSeriesIters: nsBlock.SeriesIterators.Iters(),
+		bounds:    nsBlock.Bounds,
+		indexTime: nsBlock.Bounds.Start.Add(-1 * nsBlock.Bounds.StepSize),
+	}
+}
+
+// Meta returns the metadata for the step iter
+func (m *multiSeriesBlockStepIter) Meta() block.Metadata {
+	return m.meta
+}
+
+// SeriesMeta returns metadata for the individual timeseries
+func (m *multiSeriesBlockStepIter) SeriesMeta() []block.SeriesMeta {
+	metas := make([]block.SeriesMeta, len(m.blocks))
+	for i, s := range m.blocks {
+		// todo(braskin): add name
+		metas[i].Tags = s.Metadata.Tags
+	}
+	return metas
+}
+
+// StepCount returns the total steps/columns
+func (m *multiSeriesBlockStepIter) StepCount() int {
+	if len(m.blocks) == 0 {
+		return 0
+	}
+	return m.blocks[0].Metadata.Bounds.Steps()
+}
+
+// Next moves to the next item
+func (m *multiSeriesBlockStepIter) Next() bool {
+	if len(m.seriesIters) == 0 {
+		return false
+	}
+
+	for _, s := range m.seriesIters {
+		if !s.Next() {
+			return false
+		}
+	}
+
+	m.index++
+	return true
+}
+
+// Current returns the slice of vals and timestamps for that step
+func (m *multiSeriesBlockStepIter) Current() (block.Step, error) {
+	values := make([]float64, len(m.seriesIters))
+	for i, s := range m.seriesIters {
+		values[i] = s.Current()
+	}
+
+	bounds := m.meta.Bounds
+	t := bounds.Start.Add(time.Duration(m.index) * bounds.StepSize)
+	return block.NewColStep(t, values), nil
+}
+
+// TODO: Actually free up resources
+func (m *multiSeriesBlockStepIter) Close() {}
+
+func (c *consolidatedSeriesBlockIter) Current() float64 {
+	var values []float64
+	for _, iter := range c.consolidatedNSBlockIters {
+		dp := iter.Current()
+		values = append(values, dp)
+	}
+
+	// todo(braskin): until we have consolidation
+	return values[0]
+}
+
+// Next moves to the next item
+func (c *consolidatedSeriesBlockIter) Next() bool {
+	if len(c.consolidatedNSBlockIters) == 0 {
+		return false
+	}
+
+	for _, nsBlock := range c.consolidatedNSBlockIters {
+		if !nsBlock.Next() {
+			return false
+		}
+	}
+
+	return true
+}
+
+// Close closes the underlaying iterators
+func (c *consolidatedSeriesBlockIter) Close() {
+	// todo(braskin): implement this function
+}
+
+// Next moves to the next item
+func (c *consolidatedNSBlockIter) Next() bool {
+	c.indexTime = c.indexTime.Add(c.bounds.StepSize)
```

Comment:
use index, and then do c.index++, 

indexTime := c.bounds.TimeForIndex(c.index)